### PR TITLE
Unittesting improvements

### PR DESCRIPTION
**Pull request by evilaliv3**
This pull requests apply the following improvements:
- codecoverage is now tracked using nodetests --with-coverage and results are published on coveralls.io: 
- the travisCI environment has been migrated to the [container-based infrastructure](https://docs.travis-ci.com/user/workers/container-based-infrastructure/)
- pip pachages are now cached by travis in order to speedup builds and reduce travis overheads
- the readme has been updated exposing coveralls badge

The following is the output that will be available after the merge:

https://travis-ci.org/evilaliv3/dh-virtualenv/jobs/101411288
https://coveralls.io/github/evilaliv3/dh-virtualenv

screenshot from 2016-01-10 15 19 57

screenshot from 2016-01-10 15 33 28

screenshot from 2016-01-10 15 37 56

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,5 @@
-# dh-virtualenv
-
-[![Build Status](https://travis-ci.org/spotify/dh-virtualenv.png)](https://travis-ci.org/spotify/dh-virtualenv)
+dh-virtualenv [![Build Status](https://travis-ci.org/spotify/dh-virtualenv.png)](https://travis-ci.org/spotify/dh-virtualenv) [![Coverage Status](https://coveralls.io/repos/spotify/dh-virtualenv/badge.svg?branch=unittesting-improvements&service=github)](https://coveralls.io/github/spotify/dh-virtualenv?branch=unittesting-improvements)
```
Comment (nailor) [inline README.md]: The branch here should be removed

**Comment by nailor** [comment]
Thanks for your PR! 

Seems like your custom branch went into the PR, that should probably be removed (commented the line where it is)

Ping @econchick: since you are way more proficient with coveralls stuff, could you give an eye on this change?

File: README.md (hunk 1 above)
Comment (evilaliv3) [inline README.md]: you are right! i'm going to change it to branch=master

**Comment by econchick** [comment]
Just a small thing - I'd explicitly add the `coverage` package to `travis-requirements.txt` and hard-pin the version. The `coveralls` package does depend on coverage, but doesn't hard-pin the version, and I was definitely bitten by that in the past when `coverage` came out with 4.0.

As an aside - if you find coveralls.io a bit too cumbersome, I switched to codecov.io and find it a lot easier to work with. It also comments on PRs about changes in coverage, which I really like.

**Comment by nailor** [comment]
Hi, 

since no changes happened on this PR, I'll close this. Feel free to reopen

**Pull request closed without merging**